Homework_11_Java_Circus(Exceptions_Generics_Annotations)/Homework_11_Java_Circus/src/Circus.java

```diff
@@ -0,0 +1,84 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a circus with a list of performers.
+ */
+
+public class Circus<T extends CircusPerformer> {
+	private List<T> performers;
+
+	public Circus() {
+		performers = new ArrayList<>();
+	}
+
+	public void addPerformer(T performer) {
+		performers.add(performer);
+	}
+
+	/**
+	 * Removes a performer from the circus.
+	 */
+	public T removePerformer(T performer) {
+		if (performers != null && performers.remove(performer)) {
+			return performer;
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Gets a performer by name.
+	 */
+	public T getPerformerByName(String name) {
+		for (T performer : performers) {
+			if (performer.getName().equals(name)) {
+				return performer;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Lists all performers in the circus.
+	 */
+	public List<T> listPerformers() {
+		return new ArrayList<>(performers);
+	}
+
+	/**
+	 * Saves the list of performers to a file.
+	 */
+	public void saveToFile(String filename) {
+		try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
+			for (T performer : performers) {
+				writer.write(performer.getName() + ", " + performer.getAct() + ", " + performer.getExperience());
+				writer.newLine();
+			}
+		} catch (IOException e) {
+			System.out.println("Error saving to file: " + e.getMessage());
+		}
+	}
+
+	/**
+	 * Loads the list of performers from a file.
+	 */
+	public void loadFromFile(String filename) {
+		try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
+			String line;
+			performers.clear();
```

Comment:
Nice move, makes sense as the method imitates the "import" functionality so you're making sure that the collection is empty before executing the import 🥇